internal/trie/node/copy.go

```diff
@@ -4,16 +4,27 @@
 package node
 
 // Copy deep copies the branch.
-func (b *Branch) Copy() Node {
+// Setting copyChildren to true will deep copy
+// children as well.
+func (b *Branch) Copy(copyChildren bool) Node {
```

Comment:
which are we using right now? not sure what the purpose is of having both atm?

Reply:
It's to deep copy the entire trie for tests. `copyChildren` is always false for production code though.

Reply:
hmm, I generally try not to add things to the code that's only for testing. would it make sense to just make a separate `DeepCopy` implementation?

Reply:
oh wait, i see there is, nevermind. this is fine i guess haha

Reply:
No problem, I agree with your point as well. This is used in the trie `DeepCopy()` method which is only used by tests in this package and in other packages. There are some private members in the node implementation so we need to have these in the production code for other tests in other packages to be able to deep copy. Not a big fan of that dependency, but it will do for now :wink: